py/packages/genkit/src/genkit/core/types.py

```diff
@@ -183,270 +239,370 @@ class Role(Enum):
-    media: Optional[Any] = None
-    toolRequest: Optional[Any] = None
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    text: Text | None = None
+    media: MediaModel | None = None
+    toolRequest: ToolRequest | None = None
     toolResponse: ToolResponse1
-    data: Optional[Any] = None
-    metadata: Optional[Dict[str, Any]] = None
+    data: Data | None = None
+    metadata: Metadata | None = None
 
 
-class DocumentData(BaseModel):
-    model_config = ConfigDict(extra='forbid')
-
-    content: List[
-        Union[TextPart, MediaPart, ToolRequestPart, ToolResponsePart, DataPart]
+class Part(
+    RootModel[
+        TextPart | MediaPart | ToolRequestPart | ToolResponsePart | DataPart
     ]
-    metadata: Optional[Dict[str, Any]] = None
+):
+    root: TextPart | MediaPart | ToolRequestPart | ToolResponsePart | DataPart
 
 
-class GenerateResponseChunk(BaseModel):
-    model_config = ConfigDict(extra='forbid')
+class ContentModel(RootModel[list[Part]]):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    root: list[Part]
 
-    content: List[
-        Union[TextPart, MediaPart, ToolRequestPart, ToolResponsePart, DataPart]
-    ]
-    custom: Optional[Any] = None
-    aggregated: Optional[bool] = None
-    index: float
+
+class DocumentData(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    content: list[Part]
+    metadata: dict[str, Any] | None = None
 
 
-class Message(BaseModel):
-    model_config = ConfigDict(extra='forbid')
+class GenerateResponseChunk(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    role: Role | None = None
+    index: float | None = None
+    content: list[Part]
+    custom: Any | None = None
+    aggregated: bool | None = None
+
 
+class Message(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
     role: Role
-    content: List[
-        Union[TextPart, MediaPart, ToolRequestPart, ToolResponsePart, DataPart]
-    ]
-    metadata: Optional[Dict[str, Any]] = None
+    content: list[Part]
+    metadata: dict[str, Any] | None = None
 
 
 class ModelResponseChunk(BaseModel):
-    model_config = ConfigDict(extra='forbid')
+    model_config = ConfigDict(
+        extra='forbid',
+    )
+    role: Role | None = None
+    index: Index | None = None
+    content: ContentModel
+    custom: Custom | None = None
+    aggregated: Aggregated | None = None
 
-    content: List[
-        Union[TextPart, MediaPart, ToolRequestPart, ToolResponsePart, DataPart]
-    ]
-    custom: Optional[Any] = None
-    aggregated: Optional[bool] = None
 
+class Messages(RootModel[list[Message]]):
+    root: list[Message]
 
-class Candidate(BaseModel):
-    model_config = ConfigDict(extra='forbid')
 
+class Candidate(BaseModel):
+    model_config = ConfigDict(
+        extra='forbid',
+    )
     index: float
     message: Message
-    usage: Optional[GenerationUsage] = None
+    usage: GenerationUsage | None = None
```

Comment:
just a question: is `GenerationUsage | None` preferable to `Optional[GenerationUsage]`?